src/curveToken.ts

```diff
@@ -191,6 +155,27 @@ export async function getCurveTokenInfo({
       (_, idx) => curvePool.coins(idx) as Promise<string>
     )
   );
+  const WETH = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+  const wethIdx = poolAssets.findIndex((address) => address === WETH);
+
+  if (wethIdx !== -1) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Let's add a short comment explainer for the weth logic here. Looks like if weth is detected we have to reassign the pool to something else, but I'm not sure why 😛